sdk/attestation/azure-security-attestation/src/attestation_client.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/attestation/attestation_client.hpp"
+#include <shared_mutex>
+
+#include "azure/core/base64.hpp"
+#include "azure/core/http/policies/policy.hpp"
+#include "azure/core/internal/http/pipeline.hpp"
+#include "private/attestation_client_models_private.hpp"
+#include "private/attestation_client_private.hpp"
+#include "private/attestation_common_request.hpp"
+#include "private/attestation_deserializers_private.hpp"
+#include "private/package_version.hpp"
+
+#include <string>
+
+using namespace Azure::Security::Attestation;
+using namespace Azure::Security::Attestation::Models;
+using namespace Azure::Security::Attestation::_detail;
+using namespace Azure::Security::Attestation::Models::_detail;
+using namespace Azure::Core::Http;
+using namespace Azure::Core::Http::Policies;
+using namespace Azure::Core::Http::_internal;
+
+std::string AttestationClient::ClientVersion() const { return PackageVersion::ToString(); }
+
+AttestationClient::AttestationClient(
+    std::string const& endpoint,
+    std::shared_ptr<Core::Credentials::TokenCredential const> credential,
+    AttestationClientOptions options)
+    : m_endpoint(endpoint), m_tokenValidationOptions(options.TokenValidationOptions)
+{
+  std::vector<std::unique_ptr<HttpPolicy>> perRetrypolicies;
+  if (credential)
+  {
+    //      Azure::Core::Credentials::TokenRequestContext const tokenContext
+    //          = {{_internal::UrlScope::GetScopeFromUrl(m_vaultUrl)}};
+
+    //      perRetrypolicies.emplace_back(
+    //          std::make_unique<BearerTokenAuthenticationPolicy>(credential,
+    //          std::move(tokenContext)));
+  }
+  m_apiVersion = options.Version.ToString();
+  std::vector<std::unique_ptr<HttpPolicy>> perCallpolicies;
+
+  m_pipeline = std::make_shared<Azure::Core::Http::_internal::HttpPipeline>(
+      options,
```

Comment:
the `AttestationClientOptions` contains a `TokenOptions` right?  Where are you persisting those options insde the client?

Reply:
See line 32 of attestation_client.cpp.